src/Dev/TestSession.php

```
@@ -214,7 +215,7 @@ public function submitForm(string $formID, string $button = null, array $data =
                 $formCrawler = $page->filterXPath("//form[@id='$formID']");
                 $form = $formCrawler->form();
             } catch (InvalidArgumentException $e) {
-                user_error("TestSession::submitForm failed to find the form {$formID}");
+                throw new LogicException("TestSession::submitForm failed to find the form '{$formID}'");
```

The `user_error` wasn't halting execution, which meant lines below that call methods on `$form` were trying to call methods on `null` and _that_ was the exception thrown. Better to just throw this exception directly.